File: API/Helpers/Params/UserParams.cs
using API.Models;

namespace API.Helpers.Params
{
    public class UserParams : BaseParams
    {
        public UserParams()
        {
            pageSize = 6;
            MaxPageSize = 20;
        }
        public Gender Gender { get; set; }
        public string OrderBy { get; set; }
        public string KnownAs { get; set; }
        public string City { get; set; }
        public string Country { get; set; }
        public string Province { get; set; }
        public string Focus { get; set; }
    }
}

File: client2/src/app/_model/_DTOs/UserParams.ts
import { Gender } from '../_Enum/Gender';
import { OrderByEnum } from '../_Enum/OrderByEnum';

export class UserParams {
  Gender?: Gender = null;
  OrderBy: OrderByEnum = null;
  KnownAs: string = null;
  Locations: LocationOptions[];
  Focus: string[] = [];
  PageNumber = 1;
  PageSize = 6;
  Search: string = null;
}
export class LocationOptions {
  name: string = 'Vancouver, BC Canada'; //{{City}}, {{Province}} {{Country}}
  totalResult: number = 100; //How many trainer came from this same city, province, and country
}
